deepinv/__main__.py

```diff
@@ -0,0 +1,284 @@
+# Usage: python -m deepinv train --config ./config.yaml
+
+from __future__ import annotations
+
+import deepinv as dinv
+import torch
+import torch.utils.data
+import torchvision.transforms as transforms
+
+import yaml
+from typing import Any
+from dataclasses import dataclass
+import argparse
+
+
+class CommandLineTrainer:
+    @classmethod
+    def run(cls, *, config_path: str) -> None:
+        parsed_config: cls._ParsedConfig = cls._parse_config(config_path)
+
+        device = dinv.utils.get_freer_gpu() if torch.cuda.is_available() else "cpu"
+        model = parsed_config.model
+        model.to(device)
+        physics = parsed_config.physics
+        physics.to(device)
+        optimizer = torch.optim.AdamW(model.parameters(), lr=1e-3, weight_decay=1e-8)
+        scheduler = None
+        train_dataloader = torch.utils.data.DataLoader(
+            parsed_config.dataset,
+            batch_size=4,
+            shuffle=True,
+            num_workers=2,
+        )
+        trainer = dinv.training.Trainer(
+            epochs=10,
```

Comment:
What do you think of doing a one-to-one mapping between config params and trainer options, such that this can be reduced to just like `**parsed_kwargs`?